client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/connected/ConnectedSonarLintEngine.java

```diff
@@ -61,14 +59,15 @@
   void removeStateListener(StateListener listener);
 
   /**
-   * Return rule details.
-   *
-   * @param ruleKey See {@link Issue#getRuleKey()}
-   * @return Rule details
-   * @throws IllegalArgumentException if ruleKey is unknown
-   * @since 1.2
+   * Return global rule details (not specific to any quality profile).
+   */
+  ConnectedRuleDetails getRuleDetails(String ruleKey);
+
+  /**
+   * Return rule details with specifics for a given project (severity can have been overriden in the quality profile).
+   * @param projectKey is null, the default QP will be considered
    */
-  RuleDetails getRuleDetails(String ruleKey);
+  ConnectedRuleDetails getActiveRuleDetails(@Nullable String projectKey, String ruleKey);
```

Comment:
Nitpick: I would invert the 2 parameters and put `ruleKey` first, by similarity with `getRuleDetails`.